google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -764,15 +775,33 @@ public Query orderBy(@Nonnull FieldPath fieldPath, @Nonnull Direction direction)
   }
 
   /**
-   * Creates and returns a new Query that's additionally limited to only return up to the specified
-   * number of documents.
+   * Creates and returns a new Query that only returns the first matching documents.
    *
    * @param limit The maximum number of items to return.
    * @return The created Query.
    */
   @Nonnull
   public Query limit(int limit) {
-    return new Query(firestore, options.toBuilder().setLimit(limit).build());
+    return new Query(
+        firestore, options.toBuilder().setLimit(limit).setLimitType(LimitType.First).build());
+  }
+
+  /**
+   * Creates and returns a new Query that only returns the last matching documents.
+   *
+   * <p>You must specify at least one orderBy clause for limitToLast queries, otherwise an exception
```

Comment:
. Otherwise
which exception type specifically? This might use a throws clause for that info.

Reply:
Mentioned this is an IllegalStateException.